[{"url":"https://api.github.com/repos/willmcgugan/rich/issues/comments/669597583","html_url":"https://github.com/willmcgugan/rich/issues/212#issuecomment-669597583","issue_url":"https://api.github.com/repos/willmcgugan/rich/issues/212","id":669597583,"node_id":"MDEyOklzc3VlQ29tbWVudDY2OTU5NzU4Mw==","user":{"login":"hedythedev","id":50042066,"node_id":"MDQ6VXNlcjUwMDQyMDY2","avatar_url":"https://avatars0.githubusercontent.com/u/50042066?v=4","gravatar_id":"","url":"https://api.github.com/users/hedythedev","html_url":"https://github.com/hedythedev","followers_url":"https://api.github.com/users/hedythedev/followers","following_url":"https://api.github.com/users/hedythedev/following{/other_user}","gists_url":"https://api.github.com/users/hedythedev/gists{/gist_id}","starred_url":"https://api.github.com/users/hedythedev/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/hedythedev/subscriptions","organizations_url":"https://api.github.com/users/hedythedev/orgs","repos_url":"https://api.github.com/users/hedythedev/repos","events_url":"https://api.github.com/users/hedythedev/events{/privacy}","received_events_url":"https://api.github.com/users/hedythedev/received_events","type":"User","site_admin":false},"created_at":"2020-08-05T23:32:40Z","updated_at":"2020-08-05T23:32:40Z","author_association":"CONTRIBUTOR","body":"I have also gotten this exception before when doing this on a raspberry pi, same error message. But what I did was I used `\"string\".encode('utf-8')`","performed_via_github_app":null},{"url":"https://api.github.com/repos/willmcgugan/rich/issues/comments/669825265","html_url":"https://github.com/willmcgugan/rich/issues/212#issuecomment-669825265","issue_url":"https://api.github.com/repos/willmcgugan/rich/issues/212","id":669825265,"node_id":"MDEyOklzc3VlQ29tbWVudDY2OTgyNTI2NQ==","user":{"login":"willmcgugan","id":554369,"node_id":"MDQ6VXNlcjU1NDM2OQ==","avatar_url":"https://avatars3.githubusercontent.com/u/554369?v=4","gravatar_id":"","url":"https://api.github.com/users/willmcgugan","html_url":"https://github.com/willmcgugan","followers_url":"https://api.github.com/users/willmcgugan/followers","following_url":"https://api.github.com/users/willmcgugan/following{/other_user}","gists_url":"https://api.github.com/users/willmcgugan/gists{/gist_id}","starred_url":"https://api.github.com/users/willmcgugan/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/willmcgugan/subscriptions","organizations_url":"https://api.github.com/users/willmcgugan/orgs","repos_url":"https://api.github.com/users/willmcgugan/repos","events_url":"https://api.github.com/users/willmcgugan/events{/privacy}","received_events_url":"https://api.github.com/users/willmcgugan/received_events","type":"User","site_admin":false},"created_at":"2020-08-06T09:40:39Z","updated_at":"2020-08-06T09:40:39Z","author_association":"OWNER","body":"This has come up a few times and I've advised set `PYTHONIOENCODING=utf8` which invariably fixes it.\r\n\r\nMy understanding is that when Python doesn't have a LANG env var set, it will default to ascii encoding. But Rich can't know if the encoding is a result of misconfiguration or intentional. And you'll get the same encoding error with the builtin print.\r\n\r\nA fallback is possible for many things, ascii tables, progress bars etc. But there's not much I can do if the user prints CJK or emoji. Given that the problem is a result of a misconfiguration I'm a little reluctant to spend the effort.\r\n\r\nA warning is a good idea, and I'll consider that. But I'm also thinking about wrapping that exception, and re-raising it with a helpful message about encodings.","performed_via_github_app":null},{"url":"https://api.github.com/repos/willmcgugan/rich/issues/comments/670142322","html_url":"https://github.com/willmcgugan/rich/issues/212#issuecomment-670142322","issue_url":"https://api.github.com/repos/willmcgugan/rich/issues/212","id":670142322,"node_id":"MDEyOklzc3VlQ29tbWVudDY3MDE0MjMyMg==","user":{"login":"joaqo","id":14966348,"node_id":"MDQ6VXNlcjE0OTY2MzQ4","avatar_url":"https://avatars0.githubusercontent.com/u/14966348?v=4","gravatar_id":"","url":"https://api.github.com/users/joaqo","html_url":"https://github.com/joaqo","followers_url":"https://api.github.com/users/joaqo/followers","following_url":"https://api.github.com/users/joaqo/following{/other_user}","gists_url":"https://api.github.com/users/joaqo/gists{/gist_id}","starred_url":"https://api.github.com/users/joaqo/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/joaqo/subscriptions","organizations_url":"https://api.github.com/users/joaqo/orgs","repos_url":"https://api.github.com/users/joaqo/repos","events_url":"https://api.github.com/users/joaqo/events{/privacy}","received_events_url":"https://api.github.com/users/joaqo/received_events","type":"User","site_admin":false},"created_at":"2020-08-06T19:12:59Z","updated_at":"2020-08-06T19:12:59Z","author_association":"CONTRIBUTOR","body":"Hi, I am substituting `click` for `rich` in an open source library, and hit this issue when running on docker.\r\n\r\nWhat do you think is the best way to catch this exception so we can direct the user on how to solve it, or even avoid it altogether? I would have complete control over what is printed on the screen or how the code runs, I just want to make the transition from `click` to `rich` as easy as possible on users using docker.","performed_via_github_app":null},{"url":"https://api.github.com/repos/willmcgugan/rich/issues/comments/670159942","html_url":"https://github.com/willmcgugan/rich/issues/212#issuecomment-670159942","issue_url":"https://api.github.com/repos/willmcgugan/rich/issues/212","id":670159942,"node_id":"MDEyOklzc3VlQ29tbWVudDY3MDE1OTk0Mg==","user":{"login":"willmcgugan","id":554369,"node_id":"MDQ6VXNlcjU1NDM2OQ==","avatar_url":"https://avatars3.githubusercontent.com/u/554369?v=4","gravatar_id":"","url":"https://api.github.com/users/willmcgugan","html_url":"https://github.com/willmcgugan","followers_url":"https://api.github.com/users/willmcgugan/followers","following_url":"https://api.github.com/users/willmcgugan/following{/other_user}","gists_url":"https://api.github.com/users/willmcgugan/gists{/gist_id}","starred_url":"https://api.github.com/users/willmcgugan/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/willmcgugan/subscriptions","organizations_url":"https://api.github.com/users/willmcgugan/orgs","repos_url":"https://api.github.com/users/willmcgugan/repos","events_url":"https://api.github.com/users/willmcgugan/events{/privacy}","received_events_url":"https://api.github.com/users/willmcgugan/received_events","type":"User","site_admin":false},"created_at":"2020-08-06T19:53:11Z","updated_at":"2020-08-06T19:53:11Z","author_association":"OWNER","body":"Catching the exception will be easy. There is only a single line in Console that actually writes to stdout. I can catch the exception there and re-raise it with some helpful info.\r\n\r\nThere's not much I can do to avoid it entirely that doesn't potentially break something else. ","performed_via_github_app":null},{"url":"https://api.github.com/repos/willmcgugan/rich/issues/comments/671399010","html_url":"https://github.com/willmcgugan/rich/issues/212#issuecomment-671399010","issue_url":"https://api.github.com/repos/willmcgugan/rich/issues/212","id":671399010,"node_id":"MDEyOklzc3VlQ29tbWVudDY3MTM5OTAxMA==","user":{"login":"willmcgugan","id":554369,"node_id":"MDQ6VXNlcjU1NDM2OQ==","avatar_url":"https://avatars3.githubusercontent.com/u/554369?v=4","gravatar_id":"","url":"https://api.github.com/users/willmcgugan","html_url":"https://github.com/willmcgugan","followers_url":"https://api.github.com/users/willmcgugan/followers","following_url":"https://api.github.com/users/willmcgugan/following{/other_user}","gists_url":"https://api.github.com/users/willmcgugan/gists{/gist_id}","starred_url":"https://api.github.com/users/willmcgugan/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/willmcgugan/subscriptions","organizations_url":"https://api.github.com/users/willmcgugan/orgs","repos_url":"https://api.github.com/users/willmcgugan/repos","events_url":"https://api.github.com/users/willmcgugan/events{/privacy}","received_events_url":"https://api.github.com/users/willmcgugan/received_events","type":"User","site_admin":false},"created_at":"2020-08-10T14:44:37Z","updated_at":"2020-08-10T14:44:37Z","author_association":"OWNER","body":"This is what happens if you get a unicode error now:\r\n\r\n```\r\nUnicodeEncodeError: 'ascii' codec can't encode characters in position 0-75: ordinal not in range(128)\r\n*** You may need to add PYTHONIOENCODING=utf-8 to your environment ***\r\n```\r\n\r\nHopefully that will save time for developers running Rich code in docker / CI etc.\r\n\r\nStill considering other options for this issue...","performed_via_github_app":null},{"url":"https://api.github.com/repos/willmcgugan/rich/issues/comments/671465302","html_url":"https://github.com/willmcgugan/rich/issues/212#issuecomment-671465302","issue_url":"https://api.github.com/repos/willmcgugan/rich/issues/212","id":671465302,"node_id":"MDEyOklzc3VlQ29tbWVudDY3MTQ2NTMwMg==","user":{"login":"joaqo","id":14966348,"node_id":"MDQ6VXNlcjE0OTY2MzQ4","avatar_url":"https://avatars0.githubusercontent.com/u/14966348?v=4","gravatar_id":"","url":"https://api.github.com/users/joaqo","html_url":"https://github.com/joaqo","followers_url":"https://api.github.com/users/joaqo/followers","following_url":"https://api.github.com/users/joaqo/following{/other_user}","gists_url":"https://api.github.com/users/joaqo/gists{/gist_id}","starred_url":"https://api.github.com/users/joaqo/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/joaqo/subscriptions","organizations_url":"https://api.github.com/users/joaqo/orgs","repos_url":"https://api.github.com/users/joaqo/repos","events_url":"https://api.github.com/users/joaqo/events{/privacy}","received_events_url":"https://api.github.com/users/joaqo/received_events","type":"User","site_admin":false},"created_at":"2020-08-10T16:45:27Z","updated_at":"2020-08-10T16:45:27Z","author_association":"CONTRIBUTOR","body":"That looks awesome, more than good enough for my case. Thanks!","performed_via_github_app":null},{"url":"https://api.github.com/repos/willmcgugan/rich/issues/comments/671468682","html_url":"https://github.com/willmcgugan/rich/issues/212#issuecomment-671468682","issue_url":"https://api.github.com/repos/willmcgugan/rich/issues/212","id":671468682,"node_id":"MDEyOklzc3VlQ29tbWVudDY3MTQ2ODY4Mg==","user":{"login":"DonBraulio","id":12506292,"node_id":"MDQ6VXNlcjEyNTA2Mjky","avatar_url":"https://avatars0.githubusercontent.com/u/12506292?v=4","gravatar_id":"","url":"https://api.github.com/users/DonBraulio","html_url":"https://github.com/DonBraulio","followers_url":"https://api.github.com/users/DonBraulio/followers","following_url":"https://api.github.com/users/DonBraulio/following{/other_user}","gists_url":"https://api.github.com/users/DonBraulio/gists{/gist_id}","starred_url":"https://api.github.com/users/DonBraulio/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/DonBraulio/subscriptions","organizations_url":"https://api.github.com/users/DonBraulio/orgs","repos_url":"https://api.github.com/users/DonBraulio/repos","events_url":"https://api.github.com/users/DonBraulio/events{/privacy}","received_events_url":"https://api.github.com/users/DonBraulio/received_events","type":"User","site_admin":false},"created_at":"2020-08-10T16:52:32Z","updated_at":"2020-08-10T16:52:32Z","author_association":"NONE","body":"This is great @willmcgugan ! I'll take the chance to thank you for the awesome work on this library 🙌 🙌 🙌 \r\nFeel free to close this issue when you want to.","performed_via_github_app":null}]